src/Microsoft.Azure.EventHubs/Primitives/RetryPolicy.cs

```diff
@@ -64,7 +66,11 @@ public static bool IsRetryableException(Exception exception)
             {
                 return ((EventHubsException)exception).IsTransient;
             }
-            else if (exception is OperationCanceledException)
+
+            // Other retryable exceptions here.
+            else if (exception is OperationCanceledException ||
+                exception is SocketException ||
+                exception is TaskCanceledException)
```

Comment:
>TaskCanceledException [](start = 29, length = 21)

For TaskCanceledException (and probably AggregateException) I think we should looked at the innerException if one exists to make sure it is not because of something non-transient?

Reply:
Also consider adding AggregateException which I think you can flatten and parse the innerException.

---
In reply to: [196871128](https://github.com/Azure/azure-event-hubs-dotnet/pull/289#discussion_r196871128) [](ancestors = 196871128)